go/tasks/v1/flytek8s/plugin_executor.go

```diff
@@ -283,10 +283,20 @@ func (e *K8sTaskExecutor) CheckTaskStatus(ctx context.Context, taskCtx types.Tas
 	// This must happen after sending admin event. It's safe against partial failures because if the event failed, we will
 	// simply retry in the next round. If the event succeeded but this failed, we will try again the next round to send
 	// the same event (idempotent) and then come here again...
-	if finalStatus.Phase.IsTerminal() && len(o.GetFinalizers()) > 0 {
-		err = e.ClearFinalizers(ctx, o)
-		if err != nil {
-			return types.TaskStatusUndefined, err
+	if finalStatus.Phase.IsTerminal() {
+		if en(o.GetFinalizers()) > 0 {
+			err = e.ClearFinalizers(ctx, o)
+			if err != nil {
+				return types.TaskStatusUndefined, err
+			}
+		}
+
+		// kill the object execution if still live
+		if e.handler.GetProperties().DeleteResourceOnAbort {
```

Comment:
i dont think we can do this, as this removes the idempotency property. We can do this in case of failure only if at all as it may result in an earlier failure

Reply:
updated the PR. I am going to add tests in a bit. Meanwhile, can you look at it?

Reply:
there is still an issue, dont look at it yet.

Reply:
this should work. I was worried if CheckNodestatus will even call task if task is successful. It seems it will so should be good.
https://github.com/lyft/flytepropeller/blob/master/pkg/controller/nodes/executor.go#L155